src/olympia/versions/views.py

```diff
@@ -57,13 +57,10 @@ def update_info_redirect(request, version_id):
     )
 
 
-# Should accept junk at the end for filename goodness.
 @non_atomic_requests
-def download_file(request, file_id, type=None, file_=None, addon=None):
+def download_file(request, file_id, type_=None, filename=None):
```

Comment:
I had to re-read over the patch a few times to confirm to myself that we don't actually do anything with the `filename` parameter even though we pass it down everywhere.  Do we need to specify the kwarg? (could just be `**kwargs` if we are forced to capture it in the regex to name it).

Reply:
We don't need it, we are forced to capture it to name it and pass it to `reverse()` without doing the weird string concatenation we were doing before, but you're right I could just accept `kwargs` instead here.

My initial idea was to use it to ensure the filename received was not garbage and matched the database, but that is a somewhat backwards-incompatible change as https://addons-server.readthedocs.io/en/latest/topics/api/signing.html#downloading-signed-files mentions filename `(...) is just a convenience for clients so that they write meaningful file names to disk`.